dp/pipeline/epc_packet_framework.c

```diff
@@ -84,6 +87,37 @@ static void *dp_zmq_thread(__rte_unused void *arg)
 	return NULL; //GCC_Security flag
 }
 
+#if defined (DP_BUILD) && defined (ZMQ_COMM) && defined(MULTI_UPFS)
+static void
+send_dp_credentials(void)
+{
+	static const int LEN = 3; /* strlen("ACK") */
+	static char addr_string[128] = {0};
+	char buffer[LEN];
+	void *context = zmq_ctx_new();
+	void *requester = zmq_socket(context, ZMQ_REQ);
+	snprintf(addr_string, sizeof(addr_string),
+		 "%s://%s:%u", "tcp", inet_ntoa(cp_nb_ip), cp_nb_port);
+	fprintf(stderr, "Connecting to %s\n", addr_string);
+	if (zmq_connect(requester, addr_string) != 0) {
+		rte_exit(EXIT_FAILURE, "Failed to connect to CP!\n");
+	}
+
+	fprintf(stderr, "Sent join request. Waiting for response\n");
+
+	if (zmq_send(requester, (void *)&zmq_dp_ip, sizeof(zmq_dp_ip), 0) == -1) {
+		rte_exit(EXIT_FAILURE, "Failed to send registration request to CP!\n");
+	}
+	if (zmq_recv(requester, buffer, LEN, 0) == -1) {
```

Comment:
all zmq calls are blocking ?

Reply:
ZMQ calls are blocking by default (and are set as such over here).